systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java

```diff
@@ -0,0 +1,328 @@
+        .withName("connector-with-tar-and-jar")
+        .withArtifacts(
+            new JarArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                .withNewSha512sum("b7da48d5ecd1e4199886d169ced1bf702ffbdfd704d69e0da97e78ff63c1bcece2f59c2c6c751f9c20be73472b8cb6a31b6fd4f75558c1cb9d96daa9e9e603d2")
+                .build(),
+            new TgzArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/archive/1.1.0.tar.gz")
+                .withNewSha512sum("5318b1f031d4e5eeab6f8b774c76de297237574fc51d1e81b03a10e0b5d5435a46a108b85fdb604c644529f38830ae83239c17b6ec91c90a60ac790119bb2950")
+                .build())
+        .build();
+
+    @Test
+    void testBuildFailsWithWrongChecksumOfArtifact() {
+        Plugin pluginWithWrongChecksum = new PluginBuilder()
+            .withName("connector-with-empty-checksum")
+            .withArtifacts(new JarArtifactBuilder()
+                .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                .withNewSha512sum("ads")
+                .build())
+            .build();
+
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.kafkaConnect(clusterName, 1)
+            .editOrNewSpec()
+                .withNewBuild()
+                    .withPlugins(pluginWithWrongChecksum)
+                    .withNewDockerOutput()
+                        .withNewImage(imageName)
+                    .endDockerOutput()
+                .endBuild()
+            .endSpec()
+            .build());
+
+        KafkaConnectUtils.waitForConnectNotReady(clusterName);
+        KafkaConnectUtils.waitUntilKafkaConnectStatusConditionContainsMessage(clusterName, NAMESPACE, "The Kafka Connect build .* failed");
+
+        LOGGER.info("Checking if KafkaConnect status condition contains message about build failure");
+        KafkaConnect connect = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(clusterName).get();
+        Condition connectCondition = connect.getStatus().getConditions().stream().findFirst().get();
+
+        assertTrue(connectCondition.getMessage().matches("The Kafka Connect build .* failed"));
+        assertThat(connectCondition.getType(), is(NotReady.toString()));
+
+        LOGGER.info("Replacing plugin's checksum with right one");
+        KafkaConnectResource.replaceKafkaConnectResource(clusterName, kafkaConnect -> {
+            Plugin pluginWithRightChecksum = new PluginBuilder()
+                .withName("connector-with-empty-checksum")
+                .withArtifacts(new JarArtifactBuilder()
+                    .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                    .withNewSha512sum("b7da48d5ecd1e4199886d169ced1bf702ffbdfd704d69e0da97e78ff63c1bcece2f59c2c6c751f9c20be73472b8cb6a31b6fd4f75558c1cb9d96daa9e9e603d2")
```

Comment:
Maybe create some constant with checksum?

Reply:
I created constants for all the checksums, URLs etc. -> so it can be easily changed in the future. Thanks for spotting this.